modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/hazelcast/HazelcastAgent.java

```diff
@@ -362,38 +362,54 @@ public int getIndexOfLocalNode() {
 
 
     public void notifySubscriptionsChanged(ClusterNotification clusterNotification) {
-        log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
-        this.subscriptionChangedNotifierChannel.publish(clusterNotification);
+        if (this.isActive()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
+            }
+            this.subscriptionChangedNotifierChannel.publish(clusterNotification);
+        }
     }
 
 
     public void notifyQueuesChanged(ClusterNotification clusterNotification) throws AndesException {
-        log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
-        try {
-            this.queueChangedNotifierChannel.publish(clusterNotification);
-        } catch (Exception e) {
-            log.error("Error while sending queue change notification : " + clusterNotification.getEncodedObjectAsString(), e);
-            throw new AndesException("Error while sending queue change notification : " + clusterNotification.getEncodedObjectAsString(), e);
+        if (isActive()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
+            }
+            try {
+                this.queueChangedNotifierChannel.publish(clusterNotification);
+            } catch (Exception e) {
+                log.error("Error while sending queue change notification : " + clusterNotification.getEncodedObjectAsString(), e);
+                throw new AndesException("Error while sending queue change notification : " + clusterNotification.getEncodedObjectAsString(), e);
+            }
         }
     }
 
     public void notifyExchangesChanged(ClusterNotification clusterNotification) throws AndesException {
-        log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
-        try {
-            this.exchangeChangeNotifierChannel.publish(clusterNotification);
-        } catch (Exception e) {
-            log.error("Error while sending exchange change notification" + clusterNotification.getEncodedObjectAsString(), e);
-            throw new AndesException("Error while sending exchange change notification" + clusterNotification.getEncodedObjectAsString(), e);
+        if (isActive()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Sending GOSSIP: " + clusterNotification.getDescription());
+            }
+            try {
+                this.exchangeChangeNotifierChannel.publish(clusterNotification);
+            } catch (Exception e) {
```

Comment:
Why are we catching Exception here. It may includes runtime exceptions as well. Is there a special reason to do that?